jscomp/others/js_math.ml

```diff
@@ -22,199 +22,605 @@
+  ```
+*)
 external log2 : float -> float = "log2" [@@bs.val] [@@bs.scope "Math"]
 
-(** max value *)
+(**
+  Returns the maximum of its two integer arguments.  See
+  [`Math.max`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)
+  on MDN.
+*)
 external max_int : int -> int -> int = "max" [@@bs.val] [@@bs.scope "Math"]
 
-(** max value *)
+(**
+  Returns the maximum of the integers in the given array.  See
+  [`Math.max`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)
+  on MDN.
+*)
 external maxMany_int : int array -> int = "max" [@@bs.val] [@@bs.splice] [@@bs.scope "Math"]
 
-(** max value *)
+(**
+  Returns the maximum of its two floating point arguments. See
+  [`Math.max`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)
+  on MDN.
+*)
 external max_float : float -> float -> float = "max" [@@bs.val] [@@bs.scope "Math"]
 
-(** max value *)
+(**
+  Returns the maximum of the floating point values in the given array. See
+  [`Math.max`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)
+  on MDN.
+*)
 external maxMany_float : float array -> float = "max" [@@bs.val] [@@bs.splice] [@@bs.scope "Math"]
 
-(** min value *)
+(**
+  Returns the minimum of its two integer arguments. See
+  [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min)
+  on MDN.
+*)
 external min_int : int -> int -> int = "min" [@@bs.val] [@@bs.scope "Math"]
 
-(** min value *)
+(**
+  Returns the minimum of the integers in the given array. See
+  [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min)
+  on MDN.
+*)
 external minMany_int : int array -> int = "min" [@@bs.val] [@@bs.splice] [@@bs.scope "Math"]
 
-(** min value *)
+(**
+  Returns the minimum of its two floating point arguments. See
+  [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min)
+  on MDN.
+*)
 external min_float : float -> float -> float = "min" [@@bs.val] [@@bs.scope "Math"]
 
-(** min value *)
+(**
+  Returns the minimum of the floating point values in the given array. See
+  [`Math.min`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min)
+  on MDN.
+*)
 external minMany_float : float array -> float = "min" [@@bs.val] [@@bs.splice] [@@bs.scope "Math"]
 
-(** base to the power of the exponent *)
+(**
+  Raises the given base to the given exponent. (Arguments and result are
+  integers.) See
+  [`Math.pow`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/pow)
+  on MDN.
+
+  ```res example
+  Js.Math.pow_int(~base=3, ~exp=4) == 81
+  ```
+*)
 external pow_int : base:int -> exp:int -> int = "pow" [@@bs.val] [@@bs.scope "Math"]
 [@@deprecated "use `power_float` instead, the return type may be not int"]
```

Comment:
Inconsistent gotcha here. The `pow_int` function is deprecated but docs in mdx do not mention it. For now, I copied the docs as is, without the deprecation notice.

However, looking at other similar cases (`unsafe_ceil_int / ceil_int`, `unsafe_round`, `unsafe_trunc`) to be consistent:

- `pow_int` should be renamed to `unsafe_pow_int`
- The documentation should include the standard explanation of range issues¹
- A new function `pow_int` which is an alias to the `unsafe_pow_int` should be introduced and deprecated immediatelly
- `pow_int` docs should say “please, use `unsafe_pow_int`

¹
> This function may  return values not representable by `int`, whose range is -2147483648 to  2147483647. This is because, in JavaScript, there are only 64-bit floating  point numbers, which can represent integers in the range ±(2<sup>53</sup>-1)  exactly.

Should I do it now or it is a task for the latter, when the whole docs PR settles down?

Reply:
maybe let's have a separate issue and tackle this after the transition? There shouldn't be any semantic changes.

Reply:
Sure. So for now, I guess the PR is ready for merge.